cmd/priv_val_server/main.go

```diff
@@ -32,37 +56,89 @@ func main() {
 		"chainID", *chainID,
 		"privKeyPath", *privValKeyPath,
 		"privStatePath", *privValStatePath,
+		"insecure", *insecure,
+		"withCert", *withCert,
+		"withKey", *withKey,
+		"rootCA", *rootCA,
 	)
 
 	pv := privval.LoadFilePV(*privValKeyPath, *privValStatePath)
 
-	var dialer privval.SocketDialer
-	protocol, address := tmnet.ProtocolAndAddress(*addr)
-	switch protocol {
-	case "unix":
-		dialer = privval.DialUnixFn(address)
-	case "tcp":
-		connTimeout := 3 * time.Second // TODO
-		dialer = privval.DialTCPFn(address, connTimeout, ed25519.GenPrivKey())
-	default:
-		logger.Error("Unknown protocol", "protocol", protocol)
-		os.Exit(1)
+	opts := []grpc.ServerOption{}
+	if !*insecure {
+		certificate, err := tls.LoadX509KeyPair(
+			*withCert,
+			*withKey,
+		)
+		if err != nil {
+			panic(err)
```

Comment:
nit: let's add context for better debugability

```suggestion
			panic(fmt.Errorf("failed to load X509 key pair: %w", err))
```